pymongo/message.py

```diff
@@ -878,6 +888,368 @@ def _do_batched_op_msg(
     return _batched_op_msg(operation, command, docs, ack, opts, ctx)
 
 
+class _ClientBulkWriteContext:
```

Comment:
This class has a lot of crossover with the existing `_BulkWriteContext`. Would a parent class with the common functionality to reduce code duplication make sense here?